x-pack/legacy/plugins/maps/public/connected_components/layer_panel/filter_editor/filter_editor.js

```diff
@@ -92,29 +93,35 @@ export class FilterEditor extends Component {
         anchorPosition="leftCenter"
       >
         <div className="mapFilterEditor" data-test-subj="mapFilterEditor">
-          <SearchBar
-            uiSettings={settings}
-            showFilterBar={false}
-            showDatePicker={false}
-            showQueryInput={true}
-            query={
-              layerQuery
-                ? layerQuery
-                : { language: settings.get('search:queryLanguage'), query: '' }
-            }
-            onQuerySubmit={this._onQueryChange}
-            appName="maps"
-            indexPatterns={this.state.indexPatterns}
-            store={localStorage}
-            customSubmitButton={
-              <EuiButton fill data-test-subj="mapFilterEditorSubmitButton">
-                <FormattedMessage
-                  id="xpack.maps.layerPanel.filterEditor.queryBarSubmitButtonLabel"
-                  defaultMessage="Set filter"
-                />
-              </EuiButton>
-            }
-          />
+          <KibanaContextProvider
```

Comment:
Maybe here instead of importing 3 things: (1) `SearchBar`; (2) `npStart`; (3) `KibanaContextProvider`, we could only import already connected `SearchBar` component from the data plugin.

```jsx
<npStart.plugins.data.filters.ui.SearchBar ... />
```

Reply:
In the next PR!